src/Hl7.Fhir.Support.Poco/Specification/Source/NpmPackageResolver.cs

```diff
@@ -0,0 +1,199 @@
+        }
+
+
+        private static (string url, string version) splitCanonical(string canonical)
+        {
+            if (canonical.EndsWith("|"))
+                canonical = canonical.Substring(0, canonical.Length - 1);
+
+            var position = canonical.LastIndexOf('|');
+
+            return position == -1 ?
+                (canonical, "")
+                : (canonical.Substring(0, position), canonical.Substring(position + 1));
+        }
+
+
+        ///<inheritdoc/>
+        public async Task<Resource?> ResolveByUriAsync(string uri)
+        {
+            uri.SplitLeft('/').Deconstruct(out var resource, out var id);
+
+            if (resource == null || id is null)
+                return null;
+
+            var content = await _context.GetFileContentById(resource, id);
+            return content is null ? null : toResource(content);
+        }
+
+
+        private static PackageContext createContext(string folder, bool localCache = false)
+        {
+            string? cache_folder = localCache ? folder : null;
+            var cache = new DiskPackageCache(cache_folder);
+            var project = new FolderProject(folder);
+#pragma warning disable CS8625 // Cannot convert null literal to non-nullable reference type.
+            return new PackageContext(cache, project, null);
+#pragma warning restore CS8625 // Cannot convert null literal to non-nullable reference type.
+        }
+
+        ///<inheritdoc/>
+        public IEnumerable<string> ListArtifactNames()
+        {
+            return _context.GetFileNames();
+        }
+
+        ///<inheritdoc/>
+        public Stream? LoadArtifactByName(string artifactName)
+        {
+            var content = TaskHelper.Await(() => _context.GetFileContentByFileName(artifactName));
+            return content == null ? null : new MemoryStream(Encoding.UTF8.GetBytes(content));
+        }
+
+        ///<inheritdoc/>
+        public Stream? LoadArtifactByPath(string artifactPath)
+        {
+            var content = TaskHelper.Await(() => _context.GetFileContentByFilePath(artifactPath));
+            return content == null ? null : new MemoryStream(Encoding.UTF8.GetBytes(content));
+        }
+
+        ///<inheritdoc/>
+        public IEnumerable<string> ListResourceUris(string? filter = null)
+        {
+            return _context.ListCanonicalUris(filter);
+        }
+
+        ///<inheritdoc/>
+        public async Task<Resource?> FindCodeSystemByValueSet(string valueSetUri)
+        {
+            var content = await _context.GetCodeSystemByValueSet(valueSetUri);
+            return content is null ? null : toResource(content);
+        }
+
+        ///<inheritdoc/>
+        public async Task<IEnumerable<Resource>?> FindConceptMaps(string? sourceUri = null, string? targetUri = null)
+        {
+            var content = await _context.GetConceptMapsBySourceAndTarget(sourceUri, targetUri);
+            if (content is null)
```

Comment:
You can write it like this to avoid the nullability warning:

```c#
return content is null
                ? null
                : (from item in content
                       let resource = toResource(item)
                       where resource is not null
                       select resource);
```

Reply:
Nice!